java/tools/src/java/org/apache/orc/tools/FileDump.java

```diff
@@ -293,6 +293,39 @@ private static void printMetaData(List<String> files, Configuration conf,
     }
   }
 
+  static void printTypeAnnotations(TypeDescription type, String prefix) {
+    List<String> attributes = type.getAttributeNames();
+    if (attributes.size() > 0) {
+      System.out.println("Attributes on " + prefix);
+      for(String attr: attributes) {
+        System.out.println("  " + attr + ": " + type.getAttributeValue(attr));
+      }
+    }
+    List<TypeDescription> children = type.getChildren();
+    if (children != null) {
+      switch (type.getCategory()) {
+        case STRUCT:
+          List<String> fields = type.getFieldNames();
+          for(int c = 0; c < children.size(); ++c) {
+            printTypeAnnotations(children.get(c), prefix + "." + fields.get(c));
+          }
+          break;
+        case MAP:
+          printTypeAnnotations(children.get(0), prefix + "._key");
```

Comment:
Is it guaranteed that if `children != null` the list is non-empty?

Reply:
For MAP and LIST, we can assume that they have 2 and 1 children respectively.